client/my-sites/pages/helpers.js

```diff
@@ -48,5 +48,11 @@ export const sortPagesHierarchically = ( pages ) => {
 		insertChildren( topLevelPage.ID, 1 );
 	} );
 
+	// Places the Homepage at the top of the list.
+	if ( homepageId !== 0 ) {
+		const homepage = sortedPages.findIndex( ( page ) => page.ID == homepageId );
+		sortedPages.unshift( sortedPages.splice( homepage, 1 )[ 0 ] );
```

Comment:
Should we add a safeguard for cases where `homepage` is `-1` because `findIndex` cannot find it? I think we can have some edge cases where that's possible (i.e. during pagination where there are more than 100 pages).

Reply:
Thank you! I've made those changes now, but I should note that I'm away this week, so I've committed it from my tablet. As a result, I've not been able to test the changes, so would really appreciate if you could make sure they're okay!